fetch.js

```diff
@@ -186,7 +186,9 @@
     }
 
     this.json = function() {
-      return this.text().then(JSON.parse)
+      return this.text().then(function(text) {
+        return JSON.parse(text)
+      })
```

Comment:
Looks like there's movement in Bugzilla. If you want to just wait until the issue's fixed in Firefox, that's up to you. I honestly didn't anticipate the source of the problem being an outright browser bug when I opened the PR. Otherwise, I would've just started there. Thanks for the eyes in any case!